addons/binding/org.openhab.binding.max/src/main/java/org/openhab/binding/max/internal/discovery/MaxCubeBridgeDiscovery.java

```diff
@@ -65,32 +88,36 @@ public void startScan() {
 	 * @see org.eclipse.smarthome.config.discovery.AbstractDiscoveryService#
 	 * startBackgroundDiscovery()
 	 */
-
 	@Override
 	protected void startBackgroundDiscovery() {
-		discoverCube();
+		if (cubeBackgroundDiscoveryJob == null || cubeBackgroundDiscoveryJob.isCancelled()) {
+			cubeBackgroundDiscoveryJob = scheduler.scheduleAtFixedRate(cubeDiscoveryRunnable, 0, refreshInterval, TimeUnit.SECONDS);
+		}
 	}
 
 	@Override
 	public boolean isBackgroundDiscoveryEnabled() {
```

Comment:
As it is also possible to override the 'activate' method, where the member variable is set, it should also be allowed to override isBackgroundDiscoveryEnabled(). Otherwise we have to change both methods to final. Maybe a binding developer does not want to use the AbstractDiscoveryService class through DS, but register the service manually.

I think it is a general decision, if want to use final methods or not. If we restrict methods in abstract base classes, we will loose flexibility in the implementation class. Personally I don´t like these kind of API restrictions, but maybe it makes the API easier to understand or helps to not use it wrongly.